Model/Behavior/AuditableBehavior.php

```diff
@@ -318,6 +324,10 @@ public function afterDelete(Model $Model) {
 		);
 
 		$this->Audit = ClassRegistry::init('AuditLog.Audit');
+        if(isset($this->settings[$Model->alias]['data_source'])) {
```

Comment:
The next few lines are practically a copy of the new code above. Another reason for extacting it into a generalized method.